benchmark/QR_accuracy_stability_analysis.cc

```diff
@@ -15,148 +15,317 @@
 
 template <typename T>
 static void
-test_cond_helper_0(int64_t m, 
+error_check(int64_t m,
+            int64_t n,
+            int64_t k,
+            T norm_A,
+            T* A_dat, // Original matrix reference
+            T const* A_1_dat, // Copy of the original matrix reference
+            T const* Q_dat, // Q-factor form an arbitrary QR scheme
+            T const* R_dat, // R-factor from an arbitrary QR scheme
+            T* I_ref_dat // Identity matrix reference
+            ) {
+
+    // Check orthogonality of Q
+    // Q' * Q  - I = 0
+    blas::syrk(Layout::ColMajor, Uplo::Upper, Op::Trans, k, m, 1.0, Q_dat, m, -1.0, I_ref_dat, k);
+    T norm_0 = lapack::lansy(lapack::Norm::Fro, Uplo::Upper, k, I_ref_dat, k);
+
+    // A - QR
+    blas::gemm(Layout::ColMajor, Op::NoTrans, Op::NoTrans, m, n, k, 1.0, Q_dat, m, R_dat, k, -1.0, A_dat, m);
+    
+    // Implementing max col norm metric
+    T max_col_norm = 0.0;
+    T col_norm = 0.0;
+    int max_idx = 0;
+    for(int i = 0; i < n; ++i) {
+        col_norm = blas::nrm2(m, A_dat + (m * i), 1);
```

Comment:
Please use ``&A_dat[m * i]`` instead of pointer arithmetic for improved readability.

Reply:
Fixed